### PR TITLE
BaSP-TG-05: list and delete employees

### PR DESCRIPTION
**Pull request by Maxi-SR**
- Show table with employee list - Columns: Name, Last Name, Phone, Email, Project, Status, Actions
- Add delete row functionality
- Add actions icon (delete and edit) for each row
- Warning message for delete
- Success message for confirm delete
- Redirect to add/edit form
- Add "Add new Project" button and redirect to add form

Note: When you try to assign employees to a project in the backend, it is not working, so if you wants to see the projects of an employee you have to push projects when you are creating an employee. 
Another thing, I put modals for now when you click edit or add an employee, in the next PR I'm going to redirect to another page.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
     "build": "react-scripts build",
     "lint": "eslint .",
     "lint:fix": "eslint --fix",
-    "format": "prettier --write './**/*.{js,jsx,ts,tsx,css,md,json}' --config ./.prettierrc",
+    "format": "prettier --write './**/*.{js,jsx,ts,tsx,css,md,json}' --config ./.prettierrc.json",
```
Comment (Juanoauil) [inline package.json:19]: please revert this change

File: .eslintrc.json [hunk 2]
```diff
@@ -1,34 +1,32 @@
 {
-    "settings": {
```
Comment (Juanoauil) [inline .eslintrc.json:2]: please revert this change

File: package-lock.json [hunk 3]
```diff
@@ -37247,9 +37271,11 @@
       "integrity": "sha512-0fr/mIH1dlO+x7TlcMy+bIDqKPsw/70tVyeHW787goQjhmqaZe10uwLujubK9q9Lg6Fiho1KUKDYz0Z7k7g5/g=="
     },
     "type-fest": {
-      "version": "0.21.3",
-      "resolved": "https://registry.npmjs.org/type-fest/-/type-fest-0.21.3.tgz",
-      "integrity": "sha512-t0rzBq87m3fVcduHDUFhKmyyX+9eo6WQjZvf51Ea/M0Q7+T374Jp1aUiyUl0GKxp8M/OETVHSDvmkyPgvX+X2w=="
+      "version": "0.13.1",
+      "resolved": "https://registry.npmjs.org/type-fest/-/type-fest-0.13.1.tgz",
```
Comment (Juanoauil) [inline package-lock.json:37275]: revert changes in package.json and package-lock.json

File: src/Components/Employees/Modal/index.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import styles from './modal.module.css';
+
+function Modal(props) {
```
Comment (Juanoauil) [inline src/Components/Employees/Modal/index.js:3]: you can use arrow functions

File: src/Components/Employees/index.js [hunk 5]
```diff
@@ -1,25 +1,142 @@
...
+  };
+
+  const closeModal = () => {
+    setModal(false);
+    setModalEdit(false);
+    setModalAdd(false);
+  };
+
+  const employeesToDelete = (evt) => {
+    if (evt.target.checked == true) {
```
Comment (Juanoauil) [inline src/Components/Employees/index.js:38]: if you compare, should be with ===

File: src/Components/Employees/index.js [hunk 6]
```diff
@@ -1,25 +1,142 @@
...
+    setModal(false);
+
+    const options = {
+      method: 'DELETE',
+      headers: {
+        'Content-type': 'application/json'
+      }
+    };
+    setModal(false);
+    setEmployees((current) => current.filter((x) => !checkedEmployees.includes(x._id)));
```
Comment (Juanoauil) [inline src/Components/Employees/index.js:55]: please replace checkedEmployee instead x

**Review by Juanoauil — CHANGES_REQUESTED** [review]

File: package-lock.json (hunk 3 above)
Comment (Maxi-SR) [reply package-lock.json:37275]: Damian sent to me his file

File: .eslintrc.json (hunk 2 above)
Comment (Maxi-SR) [reply .eslintrc.json:2]: Damian sent to me his file

File: package.json (hunk 1 above)
Comment (Maxi-SR) [reply package.json:19]: done

File: src/Components/Employees/index.js (hunk 6 above)
Comment (Maxi-SR) [reply src/Components/Employees/index.js:55]: Done

File: src/Components/Employees/index.js (hunk 5 above)
Comment (Maxi-SR) [reply src/Components/Employees/index.js:38]: Done

File: src/Components/Employees/Modal/modal.module.css [hunk 7]
```diff
@@ -0,0 +1,19 @@
+.container {
+  position: absolute;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 100vh;
+  width: 100%;
+  background-color: rgba(0, 0, 0, 0.5);
```
Comment (PilarFernandezC) [inline src/Components/Employees/Modal/modal.module.css:8]: Keep consistency between the use of colors. We were suggested to use color hex

File: src/Components/Employees/Modal/modal.module.css [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+  border: solid black 2px;
+}
+
+.modal {
+  width: 300px;
+  height: 130px;
+  background-color: white;
+  border: solid black 1px;
```
Comment (PilarFernandezC) [inline src/Components/Employees/Modal/modal.module.css:16]: Same here

File: src/Components/Employees/list.module.css [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+
+.item {
+  width: 300;
+  list-style: none;
+}
+
+.header {
+  display: flex;
+  justify-content: space-around;
+  background-color: rgba(55, 56, 103, 0.1);
```
Comment (PilarFernandezC) [inline src/Components/Employees/list.module.css:51]: Same here

**Review by PilarFernandezC — CHANGES_REQUESTED** [review]
Please, check all the colors

File: src/Components/Employees/Modal/modal.module.css (hunk 7 above)
Comment (Maxi-SR) [reply src/Components/Employees/Modal/modal.module.css:8]: done

File: src/Components/Employees/Modal/modal.module.css (hunk 8 above)
Comment (Maxi-SR) [reply src/Components/Employees/Modal/modal.module.css:16]: done

File: src/Components/Employees/list.module.css (hunk 9 above)
Comment (Maxi-SR) [reply src/Components/Employees/list.module.css:51]: done

File: src/Components/Employees/Modal/index.js [hunk 10]
```diff
@@ -0,0 +1,19 @@
+import styles from './modal.module.css';
+
+const Modal = (props) => {
```
Comment (DamianPalavecino) [inline src/Components/Employees/Modal/index.js:3]:
```suggestion
const Modal = ({ show, title, confirmChanges, closeModal }) => {
```

File: src/Components/Employees/index.js [hunk 11]
```diff
@@ -1,25 +1,149 @@
...
+  };
+
+  const closeModal = () => {
+    setModal(false);
+    setModalEdit(false);
+    setModalAdd(false);
+  };
+
+  const employeesToDelete = (evt) => {
+    if (evt.target.checked === true) {
```
Comment (DamianPalavecino) [inline src/Components/Employees/index.js:38]:
```suggestion
    if (evt.target.checked) {
```

File: src/Components/Employees/index.js [hunk 12]
```diff
@@ -1,25 +1,149 @@
...
+      setCheckedEmployees((current) => [...current, evt.target.id]);
+    } else {
+      setCheckedEmployees((current) => current.filter((employee) => employee !== evt.target.id));
+    }
+  };
+
+  const deleteEmployees = () => {
+    setModal(false);
+
```
Comment (DamianPalavecino) [inline src/Components/Employees/index.js:47]: Remove linebreaks inside of function

Comment (Maxi-SR) [reply src/Components/Employees/index.js:47]: done

File: src/Components/Employees/Modal/modal.module.css [hunk 13]
```diff
@@ -0,0 +1,19 @@
+.container {
+  position: absolute;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 100vh;
+  width: 100%;
+  background-color: #000000;
```
Comment (Juanoauil) [inline src/Components/Employees/Modal/modal.module.css:8]: please change to  `background-color: rgba(0, 0, 0, .5);`

Comment (Maxi-SR) [reply src/Components/Employees/Modal/modal.module.css]: done

**Review by DamianPalavecino — APPROVED** [review]
Approved, well done

**Review by Juanoauil — APPROVED** [review]

**Review by PilarFernandezC — APPROVED** [review]

**Comment by cami-diaz** [comment]
When I delete one employee works fine. But when you try to delete another one, the app crashed. I attached the screenshot

image

**Comment by cami-diaz** [comment]
The ticket says that there should be action icons (delete-edit) for each row. In this screen we only have the edit icon for each row, the delete is handle with checkboxes, should change this.

image

**Comment by cami-diaz** [comment]
The add and edit buttons should redirect to add or edit form, but for now it only show up a modal without functionality or redirect. Please fix that according to the ticket.

**Comment by cami-diaz** [comment]
The data on each row should be align with the header of each column. I attached how is display so far.

image

**Review by Juanoauil — CHANGES_REQUESTED** [review]
app crashes when we remove two employees together

**Comment by cami-diaz** [comment]
project information not visible

image

**Review by Juanoauil — APPROVED** [review]

**Pull request merged**